BASE/src/Microsoft.ApplicationInsights/Channel/IAsyncFlushable.cs

```diff
@@ -0,0 +1,25 @@
+﻿// <copyright file="IAsyncFlushable.cs" company="Microsoft">
+// Copyright © Microsoft. All Rights Reserved.
+// </copyright>
+
+namespace Microsoft.ApplicationInsights.Channel
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Represents asynchronous flush for  to application insights.
+    /// </summary>
+    public interface IAsyncFlushable : IDisposable
+    {
+        /// <summary>
+        /// Flushes the in-memory buffer asynchronously.
+        /// </summary>
+        /// <returns>
+        /// True indicates telemetry data ownership is transferred out of process, that are emitted before the flush invocation.
```

Comment:
Lets elaborate more on whats meant by out-of-process - "that data may either be sent all the way to Azure service, or stored in local disk."